### PR TITLE
fix(e2e): deletion command with `--selector` causes test flakes

### PR DESCRIPTION
**Pull request by lobkovilya**
Using `kubectl delete <resource> --selector kuma.io/mesh=<mesh>` in `AfterEach` can delete resources of the next `It()`. A more reliable approach is to list resource names and then delete them individually.

Affects:
* `MeshHTTPRoute should split traffic between internal and external services`
* `MeshTrafficPermission should allow the traffic with allow-all meshtrafficpermission`

Signed-off-by: Ilya Lobkov <ilya.lobkov@konghq.com>

### Checklist prior to review

- [X] Link to docs PR or issue --
- [X] Link to UI issue or PR --
- [X] Is the [issue worked on linked][1]? --
- [X] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [X] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Unit Tests --
- [X] E2E Tests --
- [X] Manual Universal Tests --
- [X] Manual Kubernetes Tests --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/framework/resources.go [hunk 1]
```diff
@@ -52,22 +51,23 @@ func allResourcesOfType(kumactl KumactlOptions, descriptor core_model.ResourceTy
...
-			itemDelArgs := append(args, item.GetMeta().GetName())
-			if err := k8s.RunKubectlE(cluster.GetTesting(), cluster.GetKubectlOptions(), itemDelArgs...); err != nil {
+	list, err := allResourcesOfType(*cluster.GetKumactlOptions(), resource, mesh)
+	if err != nil {
+		return err
+	}
+	for _, item := range list.GetItems() {
+		var itemDelArgs []string
+		if resource.IsPluginOriginated {
+			name, ns, err := util_k8s.CoreNameToK8sName(item.GetMeta().GetName())
```
Comment (lahabana) [inline test/framework/resources.go:61]: This is a sad hack! Shouldn't just be a method on resource or something like this?

Comment (lobkovilya) [reply test/framework/resources.go:61]: Hmm, it's a k8s-specific method, I'm not sure if we should bring it into the resource. Kuma core model is not aware if the resource is `Cluster` or `Namespace` scoped.

Do you have anything specific in mind?

**Comment by michaelbeaumont** [comment]
> `kubectl delete <resource> --selector kuma.io/mesh=<mesh>` in AfterEach can delete resources of the next It()

Why is this?

**Comment by lobkovilya** [comment]
It seems like I was wrong with this PR. 

The behavior in failed tests looked like the test is running without the MeshHTTPRoute policy, so I was thinking `kubectl delete meshhttproute --selector kuma.io/mesh=mesh` returns without waiting for the actual deletion. But now I'm reading docs and there is `--wait` flag which is `true` by default. 

So I guess there is another cause of flakiness. Closing this PR for now.

File: test/framework/resources.go (hunk 1 above)
Comment (lahabana) [reply test/framework/resources.go:61]: Yeah I don't really have a solution. It just feels a little odd.

**Pull request closed without merging**